cray_product_catalog/util/catalog_data_helper.py

```diff
@@ -0,0 +1,65 @@
+# MIT License
+#
+# (C) Copyright 2023 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# Contains a utility function for merging two dictionaries together.
+
+import re
+
+from cray_product_catalog.constants import (
+    PRODUCT_CM_FIELDS
+)
+
+
+def split_catalog_data(data):
+    """Split the passed data into data needed by main and product config map"""
+    all_unique_keys = set(data.keys())
+    comm_keys_bw_cms = all_unique_keys.intersection(PRODUCT_CM_FIELDS)
+
+    # If fields in PRODUCT_CM_FIELDS is not available in all unique keys return empty dict as second return value
+    if not comm_keys_bw_cms:
+        return {key: data[key] for key in all_unique_keys - PRODUCT_CM_FIELDS}, {}
+    else:
+        return {key: data[key] for key in all_unique_keys - PRODUCT_CM_FIELDS}, \
+            {key: data[key] for key in comm_keys_bw_cms}

```

Comment:
The `else` statement is superfluous, because the previous clause always returns from the function.
```suggestion
    return {key: data[key] for key in all_unique_keys - PRODUCT_CM_FIELDS}, \
           {key: data[key] for key in comm_keys_bw_cms}
```